s3store/s3store.go

```diff
@@ -547,3 +561,28 @@ func isAwsError(err error, code string) bool {
 	}
 	return false
 }
+
+func (store S3Store) CalcOptimalPartSize(size *int64) int64 {
```

Comment:
I was writing a test, which would run this function for ALL possible file sizes. Therefore I was trying to be as efficient on memory-use as possible, as these are quite a lot of test cases.